uiowa/uiowa
# Article content reference field

Relates to #5575 

# Todo
- [x] stacked cards - taking this opportunity to refactor how classes are implemented for these? https://github.com/uiowa/uiowa/pull/5623#issuecomment-1230811205
- [x] hr split
- [x] iisc https://github.com/uiowa/uiowa/issues/5657
- [x] sppa


# How to test
- Review the code.  Any suggestions to make the machine names better like `card_grid` and `related_manually_referenced_content`?  Is there any extra code that could be removed?
- `blt frontend`
- `ddev blt ds --site=sandbox.uiowa.edu`

## Test the update hook
- Go to https://sandbox.uiowa.ddev.site/admin/config/sitenow/uiowa-articles and verify the "do not display" option is gone from "Display related content" and that this site is set to "Grouped by tag".  Adjust it to display "Manually referenced content" and save. 
- Test on another site that may have had "Do not display" set. 

## Test the functionality
- Go to https://sandbox.uiowa.ddev.site/node/546/edit and find the "Related content" field in the sidebar and select some content to reference.  Read the help text and provide suggestions to help make the text better. 
- Edit or create a new article and use the new "Related Content" field under the "Relationships" tab. 
- A related content section should display a grid of cards at the bottom of the article in the "stacked" card design. 

## Test the splits
- HR - `ddev blt ds --site=hr.uiowa.edu` - it should work on v2 but just look different https://github.com/uiowa/uiowa/pull/5623#issuecomment-1238383764
- IISC - `ddev blt ds --site=iisc.uiowa.edu` it should not work here
- SPPA -`ddev blt ds --site=sppa.uiowa.edu`-  it should work!
